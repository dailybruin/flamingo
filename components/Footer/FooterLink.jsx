import * as React from "react";
/** @jsxRuntime classic */
/** @jsx jsx */
import { css, jsx } from "@emotion/react";
import * as MainSiteStyles from "../globals";

export default function FooterLink(props) {
  return (
    <a
      css={css`
        margin-right: 15px;
        text-decoration: none;
        color: inherit;

        ${MainSiteStyles.phone} {
          margin: auto;
        }
      `}
      href={props.url}
    >
      {props.text}
    </a>
  );
}
