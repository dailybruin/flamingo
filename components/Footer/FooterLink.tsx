import * as React from 'react';
import { css } from 'emotion'
import * as MainSiteStyles from '../globals'

interface FooterLinkProps {
  url: string,
  text: string
}

export default function FooterLink (props: FooterLinkProps) {
  return(
    <a
      className={css`
        margin-right: 15px;
        text-decoration: none;
        color: inherit;

        ${MainSiteStyles.mediaMobileBreakpoint} {
          margin: auto;
        }
      `}
      href={props.url}>{props.text}</a>
  )
}
