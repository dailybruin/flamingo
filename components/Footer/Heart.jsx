import * as React from "react";
/** @jsxRuntime classic */
/** @jsx jsx */
import { css, jsx } from "@emotion/react";

/**
 * Heart.
 */
export default function Heart(props) {
  return (
    <span
      css={css`
        &:hover {
          color: red;
          cursor: default;
        }
      `}
    >
      â™¥
    </span>
  );
}
