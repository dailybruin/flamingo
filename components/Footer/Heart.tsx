import * as React from 'react'
import { css } from 'emotion'

/**
 * Heart.
 */
export default function Heart(props: {}) {
  return (
    <span
      className={css`
        &:hover {
          color: red;
          cursor: default;
        }
      `}
    >
      â™¥
    </span>
  )
}
