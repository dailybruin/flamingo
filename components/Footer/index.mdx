---
name: Footer
route: /footer
---

import { Playground, Props } from 'docz'
import Footer from '.'

# Footer

Every web page needs a footer. Not to be confused with main site's [footer](/main-site-footer).

## Properties

<Props of={Footer} />

## Basic Usage
<Playground>
  <Footer
    githubLink={'https://dailybruin.com/'}
    developers={['Nathan Smith', 'Michael Zhang', 'Dustin Newman']}
  />
</Playground>
