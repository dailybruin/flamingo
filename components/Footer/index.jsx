import * as React from 'react'
/** @jsx jsx */
import { css, jsx } from '@emotion/core'
import Heart from './Heart'
import toSentence from '../../utils/toSentence'

/** A footer to go at the bottom of every page. */
class Footer extends React.Component {
  defaultProps = {
    license: License.Copyright,
  }

  render() {
    return (
      <footer
        css={css`
          display: grid;
          grid-template-columns: 1fr;
          grid-template-rows: auto auto;
          justify-content: center;
          margin: 1rem 0;
          padding: 0 0.4rem;
          text-align: center;

          @media (max-width: 600px) {
            font-size: 0.8rem;
          }
          ${this.props.style};
        `}
      >
        <div>
          <span>
            Content copyright Â© {this.props.copyrightYear} Daily Bruin.
          </span>{' '}
          {!!this.props.githubName && (
            <span
              css={css`
                @media (max-width: 600px) {
                  display: none;
                }
              `}
            >
              Site code available on{' '}
              <a
                href={`https://github.com/dailybruin/${this.props.githubName}`}
              >
                GitHub
              </a>
              {this.props.license === License.Copyright
                ? '.'
                : ` and available under the ${this.props.license}.`}
            </span>
          )}
        </div>
        <div>
          Built with Joyce's <Heart /> in Kerckhoff 118 by{' '}
          {typeof this.props.developers === 'string'
            ? this.props.developers
            : toSentence(this.props.developers)}
          .
        </div>
      </footer>
    )
  }
}

export default Footer
