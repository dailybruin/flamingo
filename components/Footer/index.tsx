import * as React from 'react'
import { css } from 'react'
import Heart from './Heart'
import toSentence from '../../utils/toSentence'

enum License {
  MIT = 'MIT License',
  AGPL = 'GNU AGPL',
  Copyright = 'Copyright',
}

/**
 * Footer properties.
 */
interface FooterProps {
  /** The name of the github project. The `https://github.com/dailybruin/` part is already given. */
  githubName?: string
  /** The name of the license of the project. */
  license: License
  /** A list of the developers who created the site. */
  developers: string | string[]
  /** The year that the story was published, e.g., 2018. */
  copyrightYear: number
  /** Custom css for the footer component */
  style?: string
}

/** A footer to go at the bottom of every page. */
class Footer extends React.Component<FooterProps> {
  public static defaultProps = {
    license: License.Copyright,
  }

  public render() {
    return (
      <footer
        className={css`
          display: grid;
          grid-template-columns: 1fr;
          grid-template-rows: auto auto;
          justify-content: center;
          margin: 1rem 0;
          padding: 0 0.4rem;
          text-align: center;

          @media (max-width: 600px) {
            font-size: 0.8rem;
          }
          ${this.props.style};
        `}
      >
        <div>
          <span>
            Content copyright © {this.props.copyrightYear} Daily Bruin.
          </span>{' '}
          {!!this.props.githubName && (
            <span
              className={css`
                @media (max-width: 600px) {
                  display: none;
                }
              `}
            >
              Site code available on{' '}
              <a
                href={`https://github.com/dailybruin/${this.props.githubName}`}
              >
                GitHub
              </a>
              {this.props.license === License.Copyright
                ? '.'
                : ` and available under the ${this.props.license}.`}
            </span>
          )}
        </div>
        <div>
          Built with Suzy’s <Heart /> in Kerckhoff 118 by{' '}
          {typeof this.props.developers === 'string'
            ? this.props.developers
            : toSentence(this.props.developers)}
          .
        </div>
      </footer>
    )
  }
}

export default Footer
