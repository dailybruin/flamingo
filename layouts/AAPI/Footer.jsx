import React from "react";
/** @jsx jsx */
import { css, jsx } from "@emotion/core";
import * as globals from "../../components/globals";
import FooterImage from './footer.jpg';
import Instagram from './icons/Instagram.png';
import Facebook from './icons/Facebook.png';
import Twitter from './icons/Twitter.png';
import TikTok from './icons/TikTok.png';
import Mail from './icons/Mail.png';
import { useNavigate } from 'react-router-dom';

function IconButton(props) {
    function handleClick() {
        window.location = 'http://dailybruin.com';
    }
    return (
        <div>
            <button
                css = {css`
                    background-color: transparent;
                    border-color: transparent;
                `}
                onClick={handleClick}
            >
                <img src={props.type}/>
            </button>
        </div>
    )
}

export default function AAPIFooter() {
    const ls = [Instagram, Facebook, Twitter, TikTok, Mail];
    return (
      <div
        css={css`
          background-image: url(${FooterImage});
          background-position: center;
          background-size: cover;
          background-repeat: no-repeat;
          height: 400px;
          width: 100vw;
        `}
      >
          <div
            css={css`
                display: flex;
                flex-direction: row;
                justify-content: center;
                padding-top: 180px;
            `}
          >
              {ls.map((e) => (
                  <IconButton type={e}/>
              ))}
          </div>
          <div
            css={css`
            display: flex;
            flex-direction: row;
            justify-content: center;
            color: white;
            padding-top: 30px;
            `}
          >
              Built with Suzy's Love in Kerchoff 118 by [developers]. Designed by [designers].
          </div>
      </div>
    )
  }