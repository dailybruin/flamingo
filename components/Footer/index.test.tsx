import * as React from 'react'
import { render, cleanup } from 'react-testing-library'
import Footer from '.'

describe('Footer', () => {
  afterEach(cleanup)

  it('Matches its snapshot', () => {
    const { container } = render(
      <Footer
        githubName="lux"
        developers={['Nathan Smith']}
        copyrightYear={2018}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
