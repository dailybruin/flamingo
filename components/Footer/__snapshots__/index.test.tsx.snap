// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Matches its snapshot 1`] = `
.emotion-2 {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 1rem 0;
  padding: 0 0.4rem;
  text-align: center;
}

@media (max-width:600px) {
  .emotion-2 {
    font-size: 0.8rem;
  }
}

@media (max-width:600px) {
  .emotion-0 {
    display: none;
  }
}

.emotion-1:hover {
  color: red;
  cursor: default;
}

<footer
  class="emotion-2"
>
  <div>
    <span>
      Content copyright © 
      2018
       Daily Bruin.
    </span>
     
    <span
      class="emotion-0"
    >
      Site code available on
       
      <a
        href="https://github.com/dailybruin/lux"
      >
        GitHub
      </a>
      .
    </span>
  </div>
  <div>
    Built with Suzy’s 
    <span
      class="emotion-1"
    >
      ♥
    </span>
     in Kerckhoff 118 by
     
    Nathan Smith
    .
  </div>
</footer>
`;
